src/lib.rs

```diff
@@ -88,6 +88,23 @@ pub use handler::*;
 ///  - For `Enum::MyVariant(param1, ..., paramN)` and `Enum::MyVariant { param1,
 ///    ..., paramN }`, the payload is `(param1, ..., paramN)` (where `N`>1).
 ///
+/// You can also go with even more complexity and write a syntax similar to rusts pattern matching
+/// system:
```

Comment:
I would like this description to be more formal. For example, the first bullet point doesn't show whether this pattern can be used like `Enum::MyVariant(SecondEnum::MyVariant1, SecondEnum::MyVariant2)`, and same for the other points.